From 1c8e4633f8b938a1aaa9a65d992c42bae46fe386 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=D0=A0=D1=83=D1=81=D0=BB=D0=B0=D0=BD=20=D0=98=D0=B6=D0=B1?=
 =?UTF-8?q?=D1=83=D0=BB=D0=B0=D1=82=D0=BE=D0=B2?= <lrn1986@gmail.com>
Date: Tue, 14 Apr 2015 11:02:02 +0000
Subject: [PATCH 17/20] [W32] Also use GL-related files when compiling with
 MinGW

  CXXLD    libwebkitgtk-3.0.la
./.libs/../source/webcore/platform/graphics/.libs/libplatform_la-graphicscontext3dprivate.o: In function `GraphicsContext3DPrivate':
/src/mingw/webkitgtk-2.4.8-1\bld/../webkitgtk-2.4.8/Source/WebCore/platform/graphics/GraphicsContext3DPrivate.cpp:59: undefined reference to `__ZN7WebCore9GLContext14sharingContextEv'
/src/mingw/webkitgtk-2.4.8-1\bld/../webkitgtk-2.4.8/Source/WebCore/platform/graphics/GraphicsContext3DPrivate.cpp:59: undefined reference to `__ZN7WebCore9GLContext22createOffscreenContextEPS0_'
./.libs/../source/webcore/platform/graphics/.libs/libplatform_la-graphicscontext3dprivate.o: In function `platformContext':
/src/mingw/webkitgtk-2.4.8-1\bld/../webkitgtk-2.4.8/Source/WebCore/platform/graphics/GraphicsContext3DPrivate.cpp:84: undefined reference to `__ZN7WebCore9GLContext10getCurrentEv'

Becuase GLContext.cpp is not used, which happens because it's under
TARGET_X11_OR_WAYLAND condition in the GNUmakefile.list.am.
Move TARGET_X11_OR_WAYLAND up a bit to free up GLContext.cpp (now only
protected by USE_OPENGL).
---
 Source/WebCore/GNUmakefile.list.am | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/Source/WebCore/GNUmakefile.list.am b/Source/WebCore/GNUmakefile.list.am
index 22956c7..443399d 100644
--- a/Source/WebCore/GNUmakefile.list.am
+++ b/Source/WebCore/GNUmakefile.list.am
@@ -6267,6 +6267,13 @@ if TARGET_X11_OR_WAYLAND
 platformgtk_sources += \
 	Source/WebCore/platform/cairo/WidgetBackingStoreCairo.h \
 	Source/WebCore/platform/cairo/WidgetBackingStoreCairo.cpp
+if USE_EGL
+webcoregtk_sources += \
+	Source/WebCore/platform/graphics/egl/GLContextEGL.cpp \
+	Source/WebCore/platform/graphics/egl/GLContextEGL.h
+endif # END USE_EGL
+endif # END TARGET_X11_OR_WAYLAND
+
 if USE_OPENGL
 platformgtk_sources += \
 	Source/WebCore/platform/graphics/GLContext.cpp \
@@ -6274,12 +6281,6 @@ platformgtk_sources += \
 	Source/WebCore/platform/gtk/RedirectedXCompositeWindow.cpp \
 	Source/WebCore/platform/gtk/RedirectedXCompositeWindow.h
 endif # END USE_OPENGL
-if USE_EGL
-webcoregtk_sources += \
-	Source/WebCore/platform/graphics/egl/GLContextEGL.cpp \
-	Source/WebCore/platform/graphics/egl/GLContextEGL.h
-endif # END USE_EGL
-endif # END TARGET_X11_OR_WAYLAND
 
 if TARGET_WIN32
 webcore_sources += \
-- 
1.8.5.3

